scrapy/utils/ftp.py

```diff
@@ -13,3 +15,20 @@ def ftp_makedirs_cwd(ftp, path, first_call=True):
         ftp.mkd(path)
         if first_call:
             ftp.cwd(path)
+
+def ftp_store_file( 
+        path, file, host ,port,
+        username, password, use_active_mode=False):
+    """Opens a FTP connection with passed credentials,sets current directory
+    to the directory extracted from given path, then uploads the file to server
+    """
+    ftp = FTP()
+    ftp.connect(host, port)
+    ftp.login(username, password)
+    if use_active_mode:
+        ftp.set_pasv(False)
+    file.seek(0)
+    dirname, filename = posixpath.split(path)
+    ftp_makedirs_cwd(ftp, dirname)
+    ftp.storbinary('STOR %s' % filename, file)
+    ftp.quit()
```

Comment:
:lipstick: It’s customary to leave a line at the end of the file, and recommended in PEP8, I believe.

Reply:
Will leave one :)